src/kubernetes_backup/kubernetes_backup/backup.py

```diff
@@ -44,27 +54,40 @@ def export(obj, export_dir, context):
     f.write('---\n')
     f.write(data.decode("utf-8"))
 
-def export_and_backup(context, k8s_cluster, basedir, to_s3, s3_bucket, s3_prefix, KUBERNETES_OBJECTS):
+def export_and_backup(context, k8s_cluster, basedir, s3, s3_bucket, s3_prefix, KUBERNETES_OBJECTS):
   ''' export kubernetes objects to yaml files and optionally back them up to S3 '''
-  if context:
-    cluster_label = context
-  else:
-    cluster_label = k8s_cluster
-
+  cluster_label = determine_cluster_label(context, k8s_cluster)
   export_dir = os.path.join(basedir, cluster_label)
   mkpath(export_dir)
 
-  print("INFO: Exporting objects from Kubernetes cluster '%s', 'default' namespace, as yaml files, to '%s'..." % (cluster_label, basedir))
+  logging.info(f"Exporting objects from Kubernetes cluster {cluster_label}, default namespace, as yaml files, to {basedir}...")
   for obj in KUBERNETES_OBJECTS:
     try:
       export(obj, export_dir, context)
     except CalledProcessError as e:
-      print("Error: exporting %s to yaml: %s" %(obj, e))
+      logging.critical(f"Failed to export {obj} to yaml: {e} Aborting.")
       shutil.rmtree(export_dir)
       sys.exit(1)
-  print("INFO: Done exporting")
+  logging.info("Done exporting")
 
-  if to_s3:
+  if s3:
     backup_to_s3(basedir, export_dir, cluster_label, s3_bucket, s3_prefix)
   else:
-    print("INFO: Skipping backup to S3. Please delete the local files when done!")
+    logging.info("Skipping backup to S3. Please delete the local files when done!")
+
+def full_s3_prefix(s3_prefix, cluster_label):
+  return f"{s3_prefix}/{cluster_label}"
+
+def prune(context, k8s_cluster, s3_bucket, s3_prefix, keepn, force):
+  ''' keep 'keepn' most recent backups and delete the rest '''
+  cluster_label = determine_cluster_label(context, k8s_cluster)
+  full_prefix = full_s3_prefix(s3_prefix, cluster_label)
+  logging.info(f"Pruning backups in s3://{s3_bucket}/{full_prefix}/ ...")
+  s3 = S3Interface(s3_bucket, prefix=full_prefix)
+  for backup_id in s3.backups()[keepn:]:
```

Comment:
a little simplistic - keeps the first n backups. ideally, the deleting does it by actual age of the backups. future improvement.